clients/src/test/java/org/apache/kafka/common/security/oauthbearer/internals/secured/AccessTokenRetrieverFactoryTest.java

```diff
@@ -55,21 +63,34 @@ public void testConfigureRefreshingFileAccessTokenRetriever() throws Exception {
     @Test
     public void testConfigureRefreshingFileAccessTokenRetrieverWithInvalidDirectory() {
         // Should fail because the parent path doesn't exist.
-        Map<String, ?> configs = getSaslConfigs(SASL_OAUTHBEARER_TOKEN_ENDPOINT_URL, new File("/tmp/this-directory-does-not-exist/foo.json").toURI().toString());
+        String file = new File("/tmp/this-directory-does-not-exist/foo.json").toURI().toString();
+        System.setProperty(ALLOWED_SASL_OAUTHBEARER_URLS_CONFIG, file);
+        Map<String, ?> configs = getSaslConfigs(SASL_OAUTHBEARER_TOKEN_ENDPOINT_URL, file);
         Map<String, Object> jaasConfig = Collections.emptyMap();
         assertThrowsWithMessage(ConfigException.class, () -> AccessTokenRetrieverFactory.create(configs, jaasConfig), "that doesn't exist");
     }
 
     @Test
     public void testConfigureRefreshingFileAccessTokenRetrieverWithInvalidFile() throws Exception {
-        // Should fail because the while the parent path exists, the file itself doesn't.
+        // Should fail because the parent path exists, the file itself doesn't.
         File tmpDir = createTempDir("this-directory-does-exist");
         File accessTokenFile = new File(tmpDir, "this-file-does-not-exist.json");
+        System.setProperty(ALLOWED_SASL_OAUTHBEARER_URLS_CONFIG, accessTokenFile.toURI().toString());
         Map<String, ?> configs = getSaslConfigs(SASL_OAUTHBEARER_TOKEN_ENDPOINT_URL, accessTokenFile.toURI().toString());
         Map<String, Object> jaasConfig = Collections.emptyMap();
         assertThrowsWithMessage(ConfigException.class, () -> AccessTokenRetrieverFactory.create(configs, jaasConfig), "that doesn't exist");
     }
 
+    @Test
+    public void testSaslOauthbearerTokenEndpointUrlIsNotAllowed() throws Exception {
+        // Should fail because the while the parent path exists, the file itself doesn't.
+        File tmpDir = createTempDir("not_allowed");
+        File accessTokenFile = new File(tmpDir, "not_allowed.json");
+        Map<String, ?> configs = getSaslConfigs(SASL_OAUTHBEARER_TOKEN_ENDPOINT_URL, accessTokenFile.toURI().toString());
+        assertThrowsWithMessage(IllegalArgumentException.class, () -> AccessTokenRetrieverFactory.create(configs, Collections.emptyMap()),
+                accessTokenFile.toURI().toString() + " is not allowed. Update System property 'org.apache.kafka.sasl.oauthbearer.allowed.urls' to allow " + accessTokenFile.toURI().toString());
```

Comment:
Not sure if we need to check the exact full error message. Maybe just checking that it includes `ALLOWED_SASL_OAUTHBEARER_URLS_CONFIG` would be enough?

Reply:
If so, same in `ConfigurationUtilsTest`.